### PR TITLE
Implement telegraf collecting stats on itself

### LINKED ISSUE #1348: Telegraf should send stats about itself
### Proposal:

I would like to see telegraf send metrics about it's own internals. This means what it prints to its logs:
- time take to send to output
- how many stats sent to output

It would also be nice for telegraf to count the number of errors it received from a particular input and send that as well. i.e. count failures
### Use case:

At my company we use telegraf extensively and would like to make sure it is running under the correct load etc. Having telegraf tell us these statistics would be nice so that we can better monitor telegraf's performance.

### PR DESCRIPTION
**Pull request by sparrc**
### Required for all PRs:

- [ ] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

closes #1348

### CODE REVIEW

File: selfstat/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# TODO
```
Comment (rud-bookbites) [inline selfstat/README.md]: Would probably be awesome to include

**Comment by rud-bookbites** [comment]
It's quite a long diff, is this commit fixing a few things and also adding new functionality? I don't know the culture of this project (sorry), but it seems this would be an easier review if chunked a bit.

**Comment by sparrc** [comment]
thanks @rud-bookbites, but I'm actually the maintainer of the project, you don't need to worry about reviewing this ;-)

File: internal/models/makemetric.go [hunk 2]
```diff
@@ -123,11 +122,9 @@ func makemetric(
 		case float64:
 			// NaNs are invalid values in influxdb, skip measurement
 			if math.IsNaN(val) || math.IsInf(val, 0) {
-				if debug {
```
Comment (dgnorton) [inline internal/models/makemetric.go]: Is it going to spam the log without the `debug` flag?

Comment (sparrc) [inline internal/models/makemetric.go]: This should have been caught in an earlier change. The `debug` flag has been removed in favor of the `D!` log prefix

Comment (dgnorton) [inline internal/models/makemetric.go]: It was caught. I commented on the old code. Just making sure this was the intended behavior.

**Pull request merged**